### PR TITLE
new serialization proposal

### PR DESCRIPTION
**Pull request by marrony**
This is a proposal to change values serialization, so, internally FaunaClient utilize a custom function to escape values by adding `@` tag, bug leave the native json serialization as is.

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
@marrony is this a break change?

**Comment by nathaniel-may** [comment]
could we see some code examples of what users could do after the change?

**Comment by aleclarson** [comment]
+1

Currently, the `Ref` type is lying about its properties, but this PR fixes that.

Is this a breaking change?

File: src/_json.js [hunk 1]
```diff
@@ -1,14 +1,64 @@
 'use strict'
 
 var values = require('./values')
+var Expr = require('./Expr')
+
+function faunaTypes(key, value) {
+  if (key === '') {
+    value = this[key]
```
Comment (aleclarson) [inline src/_json.js:8]: Why is this part necessary?

**Comment by aleclarson** [comment]
Is `@obj` an artifact of a past version? I noticed it's decoded by the parser, but never encoded by the replacer.

**Comment by doom-weaver** [comment]
closing as this is 2 years old.
